ironfist/src/cpp/gui/dialog.cpp

```diff
@@ -24,17 +25,27 @@ char* H2InputBox(char* msg, int len) {
   return res;
 }
 
-static wchar_t* ConvertNarrowToWide(const char* msg) {
-	int wideCharMsgLen = MultiByteToWideChar(CP_ACP, 0, msg, -1, NULL, 0);
-	wchar_t* wideCharMsg = (wchar_t*)ALLOC(wideCharMsgLen * sizeof(wchar_t));
-	MultiByteToWideChar(CP_ACP, 0, msg, -1, wideCharMsg, wideCharMsgLen);
-	return wideCharMsg;
+void DisplayError(const char* msg, const char* title) {
+	MessageBoxA(NULL, msg, title, MB_OK);
 }
 
-void DisplayError(const char* msg, const char* title) {
-	wchar_t* wideCharMsg = ConvertNarrowToWide(msg);
-	wchar_t* wideCharTitle = ConvertNarrowToWide(title);
-	MessageBox(NULL, (LPCWSTR)wideCharMsg, (LPCWSTR)wideCharTitle, MB_OK);
-	FREE(wideCharMsg);
-	FREE(wideCharTitle);
+void DisplayError(std::string msg, std::string title) {
+	MessageBoxA(NULL, msg.c_str(), title.c_str(), MB_OK);
+}
+
+void ErrorMapVariable(std::string mapVariableId, const std::string s2, std::string addErrorMessage) {

```

Comment:
dialog.cpp contains general utilities for displaying messages. There should be no details  These should all be put in save.cpp.

I think I'd like to spend our next session exploring the principle behind this.